api/api.service.ts

```diff
@@ -740,7 +740,8 @@ export class ApiService {
 
 
     download(managedFileId: XoManagedFileID, host?: string) {
-        window.location.href = (host || '') + '/XynaBlackEditionWebServices/io/download?p0=' + managedFileId.iD;
+        const subdirectory = getSubdirectory(environment.zeta.url);
+        window.location.href = (host || '') + '/' + subdirectory + 'download?p0=' + managedFileId.iD;
```

Comment:
Can we always assume, that environment.zeta.url is ending with '/'?

Reply:
Yes we can (at the moment). If not, many other places would crash. But I improved the `getSubDirectory` function to always end with a slash cause it could also be used with URLs different from *environment.zeta.url*.